src/flagpole/sentry_flagpole_context.py

```diff
@@ -1,18 +1,27 @@
-from typing import Any
+from typing import TypedDict
+
+from django.contrib.auth.models import AnonymousUser
 
 from flagpole.evaluation_context import ContextBuilder, EvaluationContextDict
 from sentry.models.organization import Organization
 from sentry.models.project import Project
 from sentry.models.user import User
 from sentry.services.hybrid_cloud.organization import RpcOrganization
+from sentry.services.hybrid_cloud.project import RpcProject
 from sentry.services.hybrid_cloud.user import RpcUser
 
 
 class InvalidContextDataException(Exception):
     pass
 
 
-def organization_context_transformer(data: dict[str, Any]) -> EvaluationContextDict:
+class SentryContextData(TypedDict):
+    actor: User | RpcUser | AnonymousUser | None
+    organization: Organization | RpcOrganization | None
+    project: Project | RpcProject | None
```

Comment:
are these supposed to be nullable (`| None`) or sometimes present `NotRequired[...]`? (or maybe both?)

Reply:
They need to be nullable because features can be organization or project scoped. When a feature is org scoped it doesn't have projects, and project scoped features often don't get organizations passed to them. There are also feature flag checks that have no `actor` either 🫠

Reply:
oh another question as well -- can this be a real type instead of a TypedDict? dicts are too easy to mess with / mess up and I'd feel more comfortable with a NamedTuple or some other concrete type

Reply:
We could have a more formal type for the context builder input data. I don't think having a type for the context data would work well because of the getsentry/sentry boundaries that we have to cross to be compatible with feature flag usage.

Reply:
I've converted everything to use dataclasses instead. GetSentry will have its own subclass of `SentryContextData`  with additional GetSentry only data. Let me know if you have any other suggestions for this implementation.